### PR TITLE
Changed Cutadapt to use nf-core modules

### PR DESCRIPTION
**Pull request by svarona**
Fixes issue #409 

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/viralrecon _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,9 +55,12 @@ Note, since the pipeline is now using Nextflow DSL2, each process will be run wi
 
 | Dependency | Old version | New version |
 | ---------- | ----------- | ----------- |
+| `cutadapt` |             | 4.6         |
+| `fastp`    | 0.23.2      | 0.23.4      |
 | `freyja`   |             | 1.5.0       |
 | `multiqc`  | 1.14        | 1.19        |
 | `fastp`    | 0.23.2      | 0.23.4      |
+| `cutadapt` |             | 4.6         |
```
Comment (maxulysse) [inline CHANGELOG.md]:
```suggestion
```

Duplicate

**Review by maxulysse — APPROVED** [review]

**Comment by maxulysse** [comment]
So the issue there is that we no longer have the adapter file in the script and we're trying to pass it within the meta map.
We should only pass files to modules within a channel.
I'd recommend to patch the modules upstream and keep this adapter input

File: conf/modules_illumina.config [hunk 2]
```diff
@@ -757,9 +757,19 @@ if (!params.skip_assembly) {
...
+                publishDir = [
+                    path: { "${params.outdir}/assembly/cutadapt/log" },
+                    enabled: false
+                ]
+            }
             withName: 'CUTADAPT' {
-                ext.args = '--overlap 5 --minimum-length 30 --error-rate 0.1'
+                ext.args = { [
+                    "--overlap 5 --minimum-length 30 --error-rate 0.1",
+                    "${meta.single_end}" == true ? "-g file:${meta.primers}" : "-g file:${meta.primers} -G file:${meta.primers}"
```
Comment (maxulysse) [inline conf/modules_illumina.config]: we cannot pass files here

**Review by maxulysse — CHANGES_REQUESTED** [review]
We should only pass files to modules within a channel.
I'd recommend to patch the modules upstream and keep this adapter input in cutadapt

**Comment by svarona** [comment]
> We should only pass files to modules within a channel. I'd recommend to patch the modules upstream and keep this adapter input in cutadapt

Thanks a lot @maxulysse! Just learned about this patch thing thanks to having @Daniel-VM nearby at work! I'm already working on it!!

**Review by maxulysse — APPROVED** [review]

**Pull request merged**